docs/Guides/creating_new_vendor_app.md

```diff
@@ -0,0 +1,137 @@
+
+1. Let's call the resolve method of the default implementation of this class by spring boot . (We need this because we need various configuration setup like adding the clientRegistrationId,state,redirectUrl and such. lets create a temp Oauth2AuthorizationRequest.
+
+2. The second line  gets the accountId from the request Url(This is a mandatory step when integrating with Vendasta. When creating the entry Url , The accountId should be tagged in the format `/entry/accountId` for it to work properly. This is explained better in the docs above.
+
+3. The next if statement will make sure that we redirect the user to the right location after the login. This also works hand in hand with another problem. That is to look for an active session with the client. If there is no active session with the client then this variable is null. So the auth flow happens only when there is no valid sessions. 
+
+4. The next line sets the accountId in the `state` parameter of your authUrl.
+
+5. We then set the callback url here using a session variable called `baseUrl` ( This is purely optional. An authenticated session can be validated a number of ways. We are using a simple session variable. On successful validation from Vendasta we get a request to the redirect Url . Since we don’t want to send the user to the redirect url always we use this variable. Let's look at this implementation later
+
+6. if the baseUrl already has a valid session then we just enter into the else part and make the object authorizationRequestResolver as `null` because we don't want to trigger an auth flow once authorized.
+
+7. The next few lines are self explanatory. They add the accountId to the `authUrl`. Our final Url would look something like this.
+
+![image.png](../../assets/images/guides/vendor_application/auth_url.png)
+
+
+This is the logic for `getAccountIdFromRequest` method 
+
+![image.png](../../assets/images/guides/vendor_application/accountid_from_request.png)
+
+
+### Step 6 - Adding a custom Success Handler to capture the redirect Url 
+
+The final addition to the chain will look something like this
+
+![image.png](../../assets/images/guides/vendor_application/success_handler.png)
+
+
+We provide a redirection endpoint and a custom success Handler
+
+![image.png](../../assets/images/guides/vendor_application/redirection_endpoint.png)
+
+
+The logic of the custom handler is simple and cute. 
+
+We check if the request is from the `/callback` endpoint we configured in Vendasta and then we check the session param that was added before to get the baseUrl. We also add the accountId that we got from the state param. That's it. We create a landing page for `/accountID` and we are done.
+
+### Step 7 - Landing page controller
+
+Let's write a simple html page using thymeleaf
+
+![image.png](../../assets/images/guides/vendor_application/thymeleaf_page.png)
```

Comment:
Let's maybe remove the "hey guy" from that template & screenshot... or come up with something else.

Reply:
The screenshot has been removed. I've included the following code as a snippet in the document.
```
<!DOCTYPE html>
<html xmlns:th="https://www.thymeleaf.org">
<head>
    <meta charset="UTF-8">
    <title>Title</title>
    <style>
        .content {
          text-align: center;
          padding: 5em;
          color: darkslategrey;
          line-height: 30px;
          letter-spacing: 1px;
          font-size: 20px;
        }
        .title {
          font-size: 50px;
          letter-spacing: 8px;
        }

    </style>
</head>
<body>
    <section>
        <article class="content">
            <h1 class="title">HELLO THERE!</h1>
            <p>
              Sample code for building marketplace-app. <br> It will help partners to kickstart their own app using Vendasta platform.
            </p>
            <h3>Account Id: <span th:text="${session.accountId}"></span></h3>
        </article>
    </section>
</body>
</html>
```